tests/jlm/hls/backend/rvsdg2rhls/MemoryConverterTests.cpp

```diff
@@ -360,4 +430,127 @@ TestThetaLoad()
 
   return 0;
 }
-JLM_UNIT_TEST_REGISTER("jlm/hls/backend/rvsdg2rhls/MemoryConverterTests-4", TestThetaLoad)
+JLM_UNIT_TEST_REGISTER("jlm/hls/backend/rvsdg2rhls/MemoryConverterTests-ThetaLoad", TestThetaLoad)
+
+static int
+TestThetaStore()
+{
+  using namespace jlm::llvm;
+  using namespace jlm::hls;
+
+  auto rvsdgModule = RvsdgModule::Create(jlm::util::filepath(""), "", "");
+
+  // Setup the function
+  std::cout << "Function Setup" << std::endl;
+  auto functionType = jlm::rvsdg::FunctionType::Create(
+      { jlm::rvsdg::bittype::Create(32),
+        jlm::rvsdg::bittype::Create(32),
+        jlm::rvsdg::bittype::Create(32),
+        jlm::llvm::PointerType::Create(),
+        jlm::rvsdg::bittype::Create(32),
+        MemoryStateType::Create() },
+      { MemoryStateType::Create() });
+
+  auto lambda = jlm::rvsdg::LambdaNode::Create(
+      rvsdgModule->Rvsdg().GetRootRegion(),
+      LlvmLambdaOperation::Create(functionType, "test", linkage::external_linkage));
+
+  // Theta
+  auto theta = jlm::rvsdg::ThetaNode::create(lambda->subregion());
+  // Predicate
+  auto idv = theta->AddLoopVar(lambda->GetFunctionArguments()[0]);
+  auto lvs = theta->AddLoopVar(lambda->GetFunctionArguments()[1]);
+  auto lve = theta->AddLoopVar(lambda->GetFunctionArguments()[2]);
+  jlm::rvsdg::bitult_op ult(32);
+  jlm::rvsdg::bitsgt_op sgt(32);
+  jlm::rvsdg::bitadd_op add(32);
+  jlm::rvsdg::bitsub_op sub(32);
+  auto arm = jlm::rvsdg::CreateOpNode<jlm::rvsdg::bitadd_op>({ idv.pre, lvs.pre }, 32).output(0);
+  auto cmp = jlm::rvsdg::CreateOpNode<jlm::rvsdg::bitult_op>({ arm, lve.pre }, 32).output(0);
+  auto match = jlm::rvsdg::match(1, { { 1, 1 } }, 0, 2, cmp);
+  idv.post->divert_to(arm);
+  theta->set_predicate(match);
+
+  // Store node
+  auto storeAddress = theta->AddLoopVar(lambda->GetFunctionArguments()[3]);
+  auto storeData = theta->AddLoopVar(lambda->GetFunctionArguments()[4]);
+  auto memoryStateArgument = theta->AddLoopVar(lambda->GetFunctionArguments()[5]);
+  auto storeOutput = StoreNonVolatileNode::Create(
+      storeAddress.pre,
+      storeData.pre,
+      { memoryStateArgument.pre },
+      32);
+  memoryStateArgument.post->divert_to(storeOutput[0]);
+
+  auto lambdaOutput = lambda->finalize({ theta->output(5) });
+  GraphExport::Create(*lambdaOutput, "f");
+
+  auto lambdaRegion = lambda->subregion();
+  jlm::rvsdg::view(rvsdgModule->Rvsdg(), stdout);
+
+  // Act
+  mem_sep_argument(*rvsdgModule);
+  // Assert
+  jlm::rvsdg::view(rvsdgModule->Rvsdg(), stdout);
+  auto * const entryMemoryStateSplitInput = *lambdaRegion->argument(5)->begin();
+  auto * entryMemoryStateSplitNode = jlm::rvsdg::input::GetNode(*entryMemoryStateSplitInput);
+  jlm::util::AssertedCast<const LambdaEntryMemoryStateSplitOperation>(
+      &entryMemoryStateSplitNode->GetOperation());
```

Comment:
You make an asserted cast here, but do not use the result. Did you mean a normal assert()?

Reply:
Changed